python/pyspark/sql/types.py

```diff
@@ -1432,10 +1434,20 @@ class Row(tuple):
 
     ``key in row`` will search through row keys.
 
-    Row can be used to create a row object by using named arguments,
-    the fields will be sorted by names. It is not allowed to omit
-    a named argument to represent the value is None or missing. This should be
-    explicitly set to None in this case.
+    Row can be used to create a row object by using named arguments.
+    It is not allowed to omit a named argument to represent the value is
+    None or missing. This should be explicitly set to None in this case.
+
+    NOTE: As of Spark 3.0.0, the Row field names are no longer sorted
+    alphabetically. To enable field sorting to create Rows compatible with
+    Spark 2.x, set the environment variable "PYSPARK_ROW_FIELD_SORTING_ENABLED"
```

Comment:
I'm curious when this compatibility will be matter?  If using Python >= 3.6 at Spark 3.0.0, do users need this compatibility?

Reply:
Yeah, good question. I think it's possible for even users that have Python 3.6 to have code that relied on the field names being sorted and this will break their existing code. So they still might need to set the env var until the existing code could be updated. Not sure how likely this scenario is...